File: mensa-api/src/main/java/com/example/mensa/dataclasses/enums/Location.java
package com.example.mensa.dataclasses.enums;

import java.nio.charset.StandardCharsets;

public enum Location {
    WUERZBURG("W\u00fcrzburg"),
    BAMBERG("Bamberg"),
    SCHWEINFURT("Schweinfurt"),
    ASCHAFFENBURG("Aschaffenburg"),

    INVALID("Invalid");

    private final String value;

    Location(final String newValue) {
        value = newValue;
    }

    public String getValue() {
        return value;
    }

    public String getValueFormatted() {
        return value
                .toLowerCase()
                .replace("\u00e4", "ae")
                .replace("\u00f6", "oe")
                .replace("\u00fc", "ue")
                .replace("\u00df", "ss");
    }
}

File: src/main/kotlin/domain/model/enums/Location.kt
package domain.model.enums

import java.util.*

enum class Location(private val value: String) {
    WUERZBURG("W\u00fcrzburg"),
    BAMBERG("Bamberg"),
    SCHWEINFURT("Schweinfurt"),
    ASCHAFFENBURG("Aschaffenburg"),
    INVALID("Invalid");

    val valueFormatted: String
        get() = value
            .lowercase(Locale.getDefault())
            .replace("\u00e4", "ae")
            .replace("\u00f6", "oe")
            .replace("\u00fc", "ue")
            .replace("\u00df", "ss")
}